### PR TITLE
Add warnings new in Xcode 8 beta 5.

### PR DESCRIPTION
**Pull request by calebd**
- Add `CLANG_WARN_INFINITE_RECURSION` to `Base.xcconfigs`
- Add `CLANG_WARN_SUSPICIOUS_MOVE` to `Base.xcconfigs`

### CODE REVIEW

File: Base/Common.xcconfig [hunk 1]
```diff
@@ -177,3 +177,11 @@ WARNING_CFLAGS = -Wno-error=unknown-warning-option -Wno-gcc-compat -Wno-unused-c
 // the default. It warns if the same variable is declared in two binaries that
 // are linked together.
 GCC_NO_COMMON_BLOCKS = YES
+
+// This warnings detects when a function will recursively call itself on every
+// code path though that function. More information can be found here:
+// http://lists.llvm.org/pipermail/cfe-commits/Week-of-Mon-20131216/096004.html
+CLANG_WARN_INFINITE_RECURSION = YES
```
Comment (NachoSoto) [inline Base/Common.xcconfig]: NICE

File: Base/Common.xcconfig [hunk 2]
```diff
@@ -177,3 +177,11 @@ WARNING_CFLAGS = -Wno-error=unknown-warning-option -Wno-gcc-compat -Wno-unused-c
...
+
+// This warning detects suspicious uses of std::move.
+CLANG_WARN_SUSPICIOUS_MOVE = YES
```
Comment (NachoSoto) [inline Base/Common.xcconfig]: You use C++? :) this seems super uselful too.

**Comment by NachoSoto** [comment]
Thanks!

File: Base/Common.xcconfig (hunk 2 above)
Comment (calebd) [inline Base/Common.xcconfig]: I do in one place. It looks like Xcode wants this on for all projects regardless of language. ¯\_(ツ)_/¯

**Pull request merged**